vnext/DeforkingPatches/ReactCommon/turbomodule/core/TurboModuleBinding.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "TurboModuleBinding.h"
+
+#include <stdexcept>
+#include <string>
+
+#include <ReactCommon/LongLivedObject.h>
+#include <cxxreact/SystraceSection.h>
+
+using namespace facebook;
+
+namespace facebook {
+namespace react {
+
+/**
+ * Public API to install the TurboModule system.
+ */
+TurboModuleBinding::TurboModuleBinding(
+    const TurboModuleProviderFunctionType &moduleProvider)
+    : moduleProvider_(moduleProvider) {}
+
+void TurboModuleBinding::install(
+    jsi::Runtime &runtime,
+    std::shared_ptr<TurboModuleBinding> binding) {
+  runtime.global().setProperty(
+      runtime,
+      "__turboModuleProxy",
+      jsi::Function::createFromHostFunction(
+          runtime,
+          jsi::PropNameID::forAscii(runtime, "__turboModuleProxy"),
+          1,
+          [binding](
+              jsi::Runtime &rt,
+              const jsi::Value &thisVal,
+              const jsi::Value *args,
+              size_t count) {
+            return binding->jsProxy(rt, thisVal, args, count);
+          }));
+}
+
+void TurboModuleBinding::invalidate() const {
+  // TODO (T45804587): Revisit this invalidation logic.
+  // The issue was that Promise resolve/reject functions that are invoked in
+  // some distance future might end up accessing PromiseWrapper that was already
+  // destroyed, if the binding invalidation removed it from the
+  // LongLivedObjectCollection.
+
+  // LongLivedObjectCollection::get().clear();
+}
+
+std::shared_ptr<TurboModule> TurboModuleBinding::getModule(
+    const std::string &name) {
+  std::shared_ptr<TurboModule> module = nullptr;
+  {
+    SystraceSection s("TurboModuleBinding::getModule", "module", name);
+    module = moduleProvider_(name);
+  }
+  return module;
+}
+
+jsi::Value TurboModuleBinding::jsProxy(
```

Comment:
>jsProxy [](start = 31, length = 7)

It's not obvious from the function name (jsProxy) that will return a "module".
Can we rename this function to something more descriptive? or is the name part of the RN abi?

Reply:
File is deleted.